realm/src/androidTest/java/io/realm/DynamicRealmTest.java

```diff
@@ -0,0 +1,332 @@
+
+
+import android.test.AndroidTestCase;
+
+import java.util.Date;
+
+import io.realm.entities.AllTypes;
+import io.realm.exceptions.RealmException;
+
+public class DynamicRealmTest extends AndroidTestCase {
+
+    private final static int TEST_DATA_SIZE = 10;
+    private static final String CLASS_ALL_TYPES = "AllTypes";
+    private static final String CLASS_OWNER = "Owner";
+
+    private RealmConfiguration defaultConfig;
+    private DynamicRealm realm;
+
+    @Override
+    public void setUp() {
+        defaultConfig = TestHelper.createConfiguration(getContext());
+        Realm.deleteRealm(defaultConfig);
+
+        // Initialize schema. DynamicRealm will not do that, so let a normal Realm create the file first.
+        Realm.getInstance(defaultConfig).close();
+        realm = DynamicRealm.getInstance(defaultConfig);
+    }
+
+    @Override
+    public void tearDown() {
+        if (realm != null) {
+            realm.close();
+        }
+    }
+
+    private void populateTestRealm(DynamicRealm realm, int objects) {
+        realm.beginTransaction();
+        realm.allObjects(CLASS_ALL_TYPES).clear();
+        for (int i = 0; i < objects; ++i) {
+            DynamicRealmObject allTypes = realm.createObject(CLASS_ALL_TYPES);
+            allTypes.setBoolean(AllTypes.FIELD_BOOLEAN, (i % 3) == 0);
+            allTypes.setBlob(AllTypes.FIELD_BINARY, new byte[]{1, 2, 3});
+            allTypes.setDate(AllTypes.FIELD_DATE, new Date());
+            allTypes.setDouble(AllTypes.FIELD_DOUBLE, 3.1415D + i);
+            allTypes.setFloat(AllTypes.FIELD_FLOAT, 1.234567F + i);
+            allTypes.setString(AllTypes.FIELD_STRING, "test data " + i);
+            allTypes.setLong(AllTypes.FIELD_LONG, i);
+        }
+        realm.commitTransaction();
+    }
+
+    private void populateTestRealm() {
+        populateTestRealm(realm, TEST_DATA_SIZE);
+    }
+
+
+    // Test that the SharedGroupManager is not reused across Realm/DynamicRealm on the same thread.
+    // This is done by starting a write transaction in one Realm and verifying that none of the data
+    // written (but not committed) is available in the other Realm.
+    public void testSeparateSharedGroups() {
+        Realm typedRealm = Realm.getInstance(defaultConfig);
+        DynamicRealm dynamicRealm = DynamicRealm.getInstance(defaultConfig);
+
+        assertEquals(0, typedRealm.where(AllTypes.class).count());
+        assertEquals(0, dynamicRealm.where(CLASS_ALL_TYPES).count());
+
+        typedRealm.beginTransaction();
+        try {
+            typedRealm.createObject(AllTypes.class);
+            assertEquals(1, typedRealm.where(AllTypes.class).count());
+            assertEquals(0, dynamicRealm.where(CLASS_ALL_TYPES).count());
+            typedRealm.cancelTransaction();
+        } finally {
+            typedRealm.close();
+            dynamicRealm.close();
+        }
+    }
+
+    // Test that Realms can only be deleted after all Typed and Dynamic instances are closed.
```

Comment:
Is it worth a test to swap `Realm` and `DynamicRealm`? I mean, closing `dynamicRealm` first and `realm` second.